packages/contracts-bedrock/src/L1/DelayedVetoable.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Semver } from "../universal/Semver.sol";
+
+contract DelayedVetoable is Semver {
+    /// @notice Error for when attempting to forward too early.
+    error ForwardingEarly();
+
+    /// @notice Error for the target is not set.
+    error TargetUnitialized();
+
+    /// @notice Error for unauthorized calls.
+    error Unauthorized(address expected, address actual);
+
+    /// @notice An event that is emitted when a call is initiated.
+    /// @param callHash The hash of the call data.
+    /// @param data The data of the initiated call.
+    event Initiated(bytes32 indexed callHash, bytes data);
+
+    /// @notice An event that is emitted each time a call is forwarded.
+    /// @param callHash The hash of the call data.
+    /// @param data The data forwarded to the target.
+    event Forwarded(bytes32 indexed callHash, bytes data);
+
+    /// @notice An event that is emitted each time a call is vetoed.
+    /// @param callHash The hash of the call data.
+    /// @param data The data forwarded to the target.
+    event Vetoed(bytes32 indexed callHash, bytes data);
+
+    /// @notice The address that all calls are forwarded to after the delay.
+    address internal _target;
+
+    // TODO(maurelian): move this to the new SuperChainConfig contract
+    // @notice The address that can veto a call.
+    address internal _vetoer;
+
+    // TODO(maurelian): move this to the new SuperChainConfig contract
+    // @notice The address that can initiate a call.
+    address internal _initiator;
+
+    /// @notice The time that a call was initiated.
+    mapping(bytes32 => uint256) internal _queuedAt;
+
+    // @notice The time to wait before forwarding a call.
+    uint256 internal _delay;
```

Comment:
Ability to set `_delay` by the vetoer or initiator if delay is 0

Reply:
In order to keep the contract fully transparent to the initiator, the delay can be activated by either party by submitting `null` data. Feedback welcome on that approach.

https://github.com/ethereum-optimism/optimism/blob/af3aa7be5ce3446718dc177c4e382f53190c18c6/packages/contracts-bedrock/src/L1/DelayedVetoable.sol#L126-L134